dev-client/src/context/AppPermissionsContext.tsx

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright © 2024 Technology Matters
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see https://www.gnu.org/licenses/.
+ */
+
+import {createContext, useCallback, useEffect, useState} from 'react';
+import {AppState, AppStateStatus} from 'react-native';
+
+import {
+  LocationPermissionResponse,
+  useForegroundPermissions,
+} from 'expo-location';
+
+export type ForegroundPermissionsType = [
+  LocationPermissionResponse | null, // permissions
+  (() => Promise<LocationPermissionResponse>) | null, // request
+  (() => Promise<LocationPermissionResponse>) | null, // get
+];
+
+export const ForegroundPermissionsContext =
+  createContext<ForegroundPermissionsType>([null, null, null]);
+
+export const ForegroundPermissionsProvider = ({
+  children,
+}: React.PropsWithChildren) => {
+  const [permissions, request, get] = useForegroundPermissions();
```

Comment:
thoughts: 

i'm trying to think through if there are any cracks for a permission change to slip through. it seems like in order for that to happen, the permissions would have to change:
- while the app is active but,
- not as a result of us calling this provider's "request" method

so i guess the question is, is there any reason this would happen besides our own programming error of calling expo's `useForegroundPermissions`? 

i can think of obscure hypotheticals (e.g. android and iOS both have modes where you can split the screen and open multiple apps simultaneously, so you could hypothetically change the settings while the app is inactive), but none that seem like blockers for this PR since it addresses the current open bugs and the user always has the "close and re-open" workaround

but maybe worth noting in a comment as a potential avenue of investigation if any more permissions bugs crop up?